src/test/java/hudson/plugins/resultscache/util/CacheServerCommTest.java

```diff
@@ -28,11 +29,12 @@ public void setUp() {
     @Test
     public void postAndGet() throws Exception {
         String id = UUID.randomUUID().toString();
-        uit.postCachedResult(id, Result.SUCCESS);
+        uit.postCachedResult(id, Result.SUCCESS, 5);
 
         Thread.sleep(1000);
 
-        Result result = uit.getCachedResult(id);
-        Assert.assertEquals(Result.SUCCESS, result);
+        CachedResult cachedResult = uit.getCachedResult(id);
+        Assert.assertEquals(Result.SUCCESS, cachedResult.getCachedResult());
+        Assert.assertEquals((Integer) 5, cachedResult.getBuildNumber());
```

Comment:
Is this cast needed?

Reply:
Yup. Error's without it due to "ambiguous method call" between assertEquals(Object, Object) and assertEquals(long, long).

Reply:
maybe you can use `.intValue()` on the actual parameter.

Reply:
Sure. No problem. I was just avoiding chaining calls where I could, and thought leaving it "as it came" felt better for the parameter under test.